lib/rules/max-lines/__tests__/index.js

```diff
@@ -0,0 +1,91 @@
+"use strict";
+
+const rule = require("..");
+const { messages, ruleName } = rule;
+
+testRule(rule, {
+  ruleName,
+  config: [1],
+
+  accept: [
+    {
+      code: ".a { color: red; } .b { color: green; }",
+      description: "count (1) == max (1)"
+    }
+  ],
+
+  reject: [
+    {
+      code: ".a { color: red; }\r\n.b { color: green; }",
+      description: "count (2) > max (1)",
+      message: messages.expected(1)
+    },
+    {
+      code: ".a { color: red; }\r.b { color: green; }",
+      description: "count (2) > max (1)",
+      message: messages.expected(1)
+    },
+    {
+      code: ".a { color: red; }\n.b { color: green; }",
+      description: "count (2) > max (1)",
+      message: messages.expected(1)
+    }
+  ]
+});
+
+testRule(rule, {
+  ruleName,
+  config: [2],
+
+  accept: [
+    {
+      code: ".a { color: red; } .b { color: green; }",
+      description: "count (1) < max (2)"
+    },
+    {
+      code: ".a { color: red; }\r\n.b { color: green; }",
+      description: "count (2) == max (2)"
+    },
+    {
+      code: ".a { color: red; }\r.b { color: green; }",
+      description: "count (2) == max (2)"
+    },
+    {
+      code: ".a { color: red; }\n.b { color: green; }",
+      description: "count (2) == max (2)"
+    }
+  ],
+
+  reject: [
+    {
+      code: ".a { color: red; }\r\n.b { color: green; }\r\n.c { color: blue; }",
+      description: "count (3) > max (2)",
+      message: messages.expected(2)
+    },
+    {
+      code: ".a { color: red; }\r.b { color: green; }\r.c { color: blue; }",
+      description: "count (3) > max (2)",
+      message: messages.expected(2)
+    },
+    {
+      code: ".a { color: red; }\n.b { color: green; }\n.c { color: blue; }",
+      description: "count (3) > max (2)",
+      message: messages.expected(2)
+    },
+    {
+      code: ".a {\r\ncolor: red;\r\n}",
+      description: "count (3) > max (2)",
+      message: messages.expected(2)
+    },
+    {
+      code: ".a {\rcolor: red;\r}",
+      description: "count (3) > max (2)",
+      message: messages.expected(2)
+    },
+    {
+      code: ".a {\ncolor: red;\n}",
+      description: "count (3) > max (2)",
+      message: messages.expected(2)
+    }
+  ]
+});
```

Comment:
Looks very good! Can we add test for `cssinjs` ?

Reply:
Hey! I'm not overly familiar with `cssinjs`, but I'll take a look into it now!

Reply:
You can find example in other tests

Reply:
I don't think it makes much sense to use this rule with CSS-in-JS, because .js could have many imports and logic besides styling parts.

Reply:
@hugomrdias I think it should work well for `css-in-js`, like:

max: 10

```
const Button = styled.a`
  /* This renders the buttons above... Edit me! */
  display: inline-block;
  border-radius: 3px;
  padding: 0.5rem 0;
  margin: 0.5rem 1rem;
  width: 11rem;
  background: transparent;
  color: white;
  border: 2px solid white;

  /* The GitHub button is a primary button
   * edit this to target it specifically! */
  ${props => props.primary && css`
    background: white;
    color: palevioletred;
  `}
`
```

And here we get warning because styles more than 20 lines. eslint max-lines rule for whole js file. stylelint rule only for styles blocks